### ISSUE TITLE
error occurs during training

### ISSUE THREAD

**Issue by zkkao** [Qoala-T/QC#39]
Dear Authors:

I use R version 4.1.2 and run commands line by line in the script of Qoala_T_B_subset_based_github.R. An error occurs during training step:

rf.tune = train(y=training$Rating,
                  x=subset(training, select=-c(Rating)),
                  method = "rf",
                  metric = "ROC",
                  trControl = ctrl,
                  ntree = 501,
                  tuneGrid=expand.grid(mtry = c(8)),
                  verbose=FALSE)

and the error message is:

Error in apply(testOutput[, lev], 1, function(x) x/sum(x)) : 
 dim(X) must have a positive length
Calls: train -> train.default -> evalSummaryFunction -> t -> apply
Execution halted

Is there any suggestion?

**Comment by larawierenga** [comment]
Hi there,

Ik looks like there is some missing data. Does the training set include all data and columns and aren’t there any missing values in the rating column?

Lara


On 12 Jan 2022, at 02:42, zkkao ***@***.***> wrote:

﻿

Dear Authors:

I use R version 4.1.2 and run commands line by line in the script of Qoala_T_B_subset_based_github.R. An error occurs during training step:

rf.tune = train(y=training$Rating,
x=subset(training, select=-c(Rating)),
method = "rf",
metric = "ROC",
trControl = ctrl,
ntree = 501,
tuneGrid=expand.grid(mtry = c(8)),
verbose=FALSE)

and the error message is:

Error in apply(testOutput[, lev], 1, function(x) x/sum(x)) :
dim(X) must have a positive length
Calls: train -> train.default -> evalSummaryFunction -> t -> apply
Execution halted

Is there any suggestion?

—
Reply to this email directly, view it on GitHub<https://eur03.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FQoala-T%2FQC%2Fissues%2F39&data=04%7C01%7Cl.m.wierenga%40fsw.leidenuniv.nl%7C57cfc8671f1a4bc028f608d9d56cd6a4%7Cca2a7f76dbd74ec091086b3d524fb7c8%7C0%7C0%7C637775485717479492%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=oarqK5FBNJM4Tq2sEBU3YtQv%2BvwjZx4lmGS0RGw7Ex8%3D&reserved=0>, or unsubscribe<https://eur03.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FAITGLADSMVPN3AKQAS46XO3UVTMBHANCNFSM5LXYNU2Q&data=04%7C01%7Cl.m.wierenga%40fsw.leidenuniv.nl%7C57cfc8671f1a4bc028f608d9d56cd6a4%7Cca2a7f76dbd74ec091086b3d524fb7c8%7C0%7C0%7C637775485717479492%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=%2Fmwwx0OyOUTsMfk3v03zAao9XHfaylnRwZqoki7WF0c%3D&reserved=0>.
Triage notifications on the go with GitHub Mobile for iOS<https://eur03.safelinks.protection.outlook.com/?url=https%3A%2F%2Fapps.apple.com%2Fapp%2Fapple-store%2Fid1477376905%3Fct%3Dnotification-email%26mt%3D8%26pt%3D524675&data=04%7C01%7Cl.m.wierenga%40fsw.leidenuniv.nl%7C57cfc8671f1a4bc028f608d9d56cd6a4%7Cca2a7f76dbd74ec091086b3d524fb7c8%7C0%7C0%7C637775485717479492%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=%2B9L3TtPHPa0HrHDto6zUXoHE4H086%2FD4izWFsvZay1w%3D&reserved=0> or Android<https://eur03.safelinks.protection.outlook.com/?url=https%3A%2F%2Fplay.google.com%2Fstore%2Fapps%2Fdetails%3Fid%3Dcom.github.android%26referrer%3Dutm_campaign%253Dnotification-email%2526utm_medium%253Demail%2526utm_source%253Dgithub&data=04%7C01%7Cl.m.wierenga%40fsw.leidenuniv.nl%7C57cfc8671f1a4bc028f608d9d56cd6a4%7Cca2a7f76dbd74ec091086b3d524fb7c8%7C0%7C0%7C637775485717479492%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=iOhZDTQZjSOehz7HXIH6Yy2d%2FyalrQFcvRNpEV8fY3w%3D&reserved=0>.

**Comment by zkkao** [comment]
Hi Lara,
I have confirm that there isn't any missing value in rating column. And training set include all data and columns. The dimension of my training set is [79, 186]. The version of caret is 6.0-90. Can you give some advise.

**Comment by larawierenga** [comment]
Hi,

That should be correct indeed. What fs version is the data from?

Lara



On 13 Jan 2022, at 12:41, zkkao ***@***.******@***.***>> wrote:



Hi Lara,
I have confirm that there isn't any missing value in rating column. And training set include all data and columns. The dimension of my training set is [79, 186]. The version of caret is 6.0-90. Can you give some advise.

—
Reply to this email directly, view it on GitHub<https://eur03.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FQoala-T%2FQC%2Fissues%2F39%23issuecomment-1012058861&data=04%7C01%7Cl.m.wierenga%40fsw.leidenuniv.nl%7Cdab8f889d4ee43327d6808d9d689b2f1%7Cca2a7f76dbd74ec091086b3d524fb7c8%7C0%7C0%7C637776709164358552%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=lC4CMk68Sp6Ti2DpsSkcafFG4zilQncGlA5kBaL8jYw%3D&reserved=0>, or unsubscribe<https://eur03.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FAITGLAG3BYZ3EYANPOAOEE3UV2273ANCNFSM5LXYNU2Q&data=04%7C01%7Cl.m.wierenga%40fsw.leidenuniv.nl%7Cdab8f889d4ee43327d6808d9d689b2f1%7Cca2a7f76dbd74ec091086b3d524fb7c8%7C0%7C0%7C637776709164358552%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=C3EMxh0uXi2eI%2BAWREF9QTXE3tedrdHaGvYjW5y4t14%3D&reserved=0>.
Triage notifications on the go with GitHub Mobile for iOS<https://eur03.safelinks.protection.outlook.com/?url=https%3A%2F%2Fapps.apple.com%2Fapp%2Fapple-store%2Fid1477376905%3Fct%3Dnotification-email%26mt%3D8%26pt%3D524675&data=04%7C01%7Cl.m.wierenga%40fsw.leidenuniv.nl%7Cdab8f889d4ee43327d6808d9d689b2f1%7Cca2a7f76dbd74ec091086b3d524fb7c8%7C0%7C0%7C637776709164358552%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=5uEa%2FZJ%2F0w%2BZLD8G%2FauV3P01D5wvGWV5a1FwkkvbsGA%3D&reserved=0> or Android<https://eur03.safelinks.protection.outlook.com/?url=https%3A%2F%2Fplay.google.com%2Fstore%2Fapps%2Fdetails%3Fid%3Dcom.github.android%26referrer%3Dutm_campaign%253Dnotification-email%2526utm_medium%253Demail%2526utm_source%253Dgithub&data=04%7C01%7Cl.m.wierenga%40fsw.leidenuniv.nl%7Cdab8f889d4ee43327d6808d9d689b2f1%7Cca2a7f76dbd74ec091086b3d524fb7c8%7C0%7C0%7C637776709164358552%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=7Q0nQCsqMhIU1SlfIGDiY0Agz1hFK3%2BKLlC2cYpE%2FQI%3D&reserved=0>.

**Comment by zkkao** [comment]
Hi Lara,
The MRI data was processed by Freesurfer v7.1.1.
